test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -0,0 +1,70 @@
+// Flags: --expose-internals
+import '../common/index.mjs';
+import mod from 'internal/modules/esm/loader';
+import assert from 'assert';
+
+const { ESMLoader } = mod;
+
+{
+  const esmLoader = new ESMLoader();
```

Comment:
```suggestion
{
  const esmLoader = new mod.ESMLoader();
```

Reply:
Is there a benefit to doing this instead of what's currently there? (keeping in mind there will be more blocks and ESMLoader instances added in future)